codegen/kuberesource/crd.go

```diff
@@ -153,3 +188,90 @@ func CustomResourceDefinition(
 	}
 	return crd
 }
+
+type ErrMap map[string]error
```

Comment:
i'm not sure the code here and below belong in `codegen` subpackage. i think any code that is called outside of code generation should live in a `pkg` subdir

Reply:
I've placed it it here mainly because it is related to the logic in this file. what package do you think it should go in?

Reply:
id say either create a new package like `pkg/crdutils` or use existing like `pkg/controllerutils`